lib/omniauth/strategies/cronofy.rb

```diff
@@ -28,18 +28,19 @@ def raw_info
       end
 
       def request_phase
-        link_token = session['omniauth.params']['link_token']
-        if link_token && !link_token.empty?
-          options[:authorize_params] ||= {}
-          options[:authorize_params].merge!(:link_token => link_token)
-        end
-
-        avoid_linking = session['omniauth.params']['avoid_linking']
-        if avoid_linking
-          options[:authorize_params] ||= {}
-          options[:authorize_params].merge!(:avoid_linking => avoid_linking)
+        params = {}
+        %w{
+          link_token
+          avoid_linking
+          provider_name
+        }.each do |param|
+          if session['omniauth.params'][param] && !session['omniauth.params'][param].empty?
```

Comment:
Changed behaviour slightly for `avoid_linking`, not sure there's any need for the empty check.

Could also refactor out `session_params = session['omniauth.params']`.